File: Sorting/Insertion_Sort.cpp
#include<iostream>
using namespace std;

//Time complexity : O(n^2)
//comparision sorting
//not used for large data sorting
void insertionSort(int a[],int size)
{
  int i,j,key;
  for(i=1;i<=size-1;i++)
  {
    key=a[i];
    j=i;
    while (a[j-1]>key && j>=1)
    {
      a[j]=a[j-1];
      j--;
    }
    a[j]=key;
  }
  for (int k=0;k<size;k++)
    cout<<a[k]<<"\t";
}


int main()
{
  int a[]={34,2,42,1,11,23};
  int n=sizeof(a)/sizeof(a[0]);
  insertionSort(a,n);
  return 0;
}

File: Sorting/Insertion_sort/Insertion_Sort.py

#Insertion sort is a simple sorting algorithm that builds the final sorted array one item at a time. It is much less efficient on large lists than more advanced algorithms such as quicksort, heapsort, or merge sort.

def insertion_sort(input_list):

    # Sort the elements to arrange in order using Insertion Sort algorithm
    for i in range(1, len(input_list)): # Set the range to run the loop n times
        j = i - 1
        nxt_element = input_list[i]

        # Compare the current element with next element and sorting it
        while (input_list[j] > nxt_element) and (j >= 0):
            input_list[j + 1] = input_list[j]
            j = j - 1
            input_list[j + 1] = nxt_element


input_list = [2,5,6,5,34,11,23,0,3]
insertion_sort(input_list)
print(input_list)